### ISSUE TITLE
bug: duplicate celebrations

### ISSUE THREAD

**Issue by tukusejssirs** [romcal/romcal#10]
I’ve just found out that the Sts Cyril and Methodius have two feast days in the output of `romcal`. One is according to General Roman Calendar on 14th February (which is never celebrated in Slovakia) and the other on 5th July (which is always celebrated in Slovakia, as it is a feast and public holiday in Slovakia).

Here are the lines I found:

```bash
$ cat calendars/slovakia.js | grep -nA4 -B2 Cyril
14-    var dates = [
15-      {
16:        "key": "saintsCyrilMonkAndMethodiusBishop",
17-        "type": Types[4],
18-        "moment": moment.utc({ year: arguments[0], month: 1, day: 14 }),
19-        "data": {
20-          "meta": {
--
105-      },
106-      {
107:        "key": "saintsCyrilAndMethodiusSlavicMissionaries",
108-        "type": Types[0],
109-        "moment": moment.utc({ year: arguments[0], month: 6, day: 5 }),
110-        "data": {}
111-      },
```

I suggest to remove the first one from the Slovakian calendar (`calendar/slovakia.js`).

**Comment by tukusejssirs** [comment]
Or probably I am wrong—I checked it only after I posted this issue on [this](https://lc.kbs.sk/?den=20170214) website (14th February) (which is hosted by Conference of Slovakian Bishops, KBS) and is in Slovakian language only. It states that there’s an option to celebrate a votive Holy Mass of Sts Cyril, monk, and Methodius, bishop, the patron saints of Europe (original: votívna svätá omša o svätých Cyrilovi, mníchovi, a Metodovi, biskupovi, patrónoch Európy).

Just for complete information, [this](https://lc.kbs.sk/?den=20170705) website (5th July) says that on that day is celebrated the feast of Sts Cyril and Methodius, Slavic missionaries (Svätých Cyrila a Metoda, slovanských vierozvestov).

I apologise for opening this issue.

**Comment by pejulian** [comment]
Hi @tukusejssirs,

Thanks for pointing out this duplicate!
I took a look at the resource mentioned and saw what you meant by the option for this celebration to land on 14th February.

However, this observation has alerted me to a bug in the code whereby the feast of Saints Cyril and Methodius was being wrongly represented by two unique keys which are `saintsCyrilMonkAndMethodiusBishop` and `saintsCyrilAndMethodiusSlavicMissionaries`. This is not correct as these 2 different keys will interfere with `romcal`'s logic of overriding feasts in the `general` calendar based on unique key names.

Therefore, I have removed the `saintsCyrilAndMethodiusSlavicMissionaries` key in favor of the `saintsCyrilMonkAndMethodiusBishop` in `src/calendars/slovakia.mjs` as this is the key that will correctly override the entry in `src/calendars/general.mjs` when invoking `calendarFor()` for the `sk` locale. 

Also, I have added a new boolean flag called `saintsCyrilMonkAndMethodiusBishopOnFeb14` to `src/calendars/slovakia.mjs` so that `romcal` can generate the Liturgical calendar for Slovakia with this feat on February 14 if such a use case is required. (The same logic has also been added for Czech Republic). In both calendars, the entry has been marked as a "Feast". Do let me know if it is supposed to be "Solemnity" and I can make the change.

I also added a new `src/locales/skSk.mjs` locale file for Slovakia where the key `saintsCyrilMonkAndMethodiusBishop` is localized as `Saints Cyril and Methodius, Slavic Missionaries`.

The fix is being added in the `romcal-es6` branch and will be soon merged into `master`.

Also, a humble request from me: perhaps you wouldn't mind contributing Slovakian texts to the `skSk.mjs` file so that `romcal` can generate the Liturgical Calendar for Slovakia in actual Slovak texts :)

God bless!

**Comment by tukusejssirs** [comment]
Yes, I noticed the duplicate key names, too, but forgot to include it in the issue description. ;)

On 5th July in Slovakia, it is always celebrated as feast, while on 14th Feb, the [site](https://lc.kbs.sk/?den=20170214) mentioned it only as a _**votive**_ Holy Mass (of which I didn’t know at all), therefore I presume that the mention of votive Holy Mass is only because of the memoria date in the General Roman Calendar. However according to my liturgical knowledge, a votive Holy Mass can be celebrated anytime where there is a ferial day or optional memoria, thus I think I shouldn’t be necessary to include it on 14th Feb. But I need to double-check it (14th Feb) in an officially release Directory (I don’t know how this liturgical assistant / booklet is called in English, but in Slovak it is called Direktórium [ˈdirekˌtoːrium]; it includes everything related to the Liturgies of Holy Mass and Hours, e.g. readings, Credo, Gloria, Week from 4-week cycle, etc), but I must get my hands on any but current, because this year Ash Wednesday has fallen on 14th Feb.

As for the l10n, I already started translating `locales/en.json` from `master` branch, but now I’m a bit puzzled—should I rather translate `src/locales/skSk.mjs` file from `romcal-es6` branch?

**Comment by tukusejssirs** [comment]
And according to what I found on the Internet, you probably got it right that in Czech Republic it is the same as in Slovakia: on 14th Feb Memoria of St Valentine is celebrated and on 5th July Feast of Sts Cyril and Methodius.

**Comment by pejulian** [comment]
@tukusejssirs 

Thanks for the clarification about the celebration type. It will be noted as a feast in the both the calendars.

If I am not mistaken, the same book is called the Divine Office in English. 

For 2018, Ash Wednesday would have overriden anything that coincided on 14th Feb due to its importance (priority and type). I will add a test case to ensure that this is the case in `romcal`'s output. The newly added flag should be able to cater for the votive possibility of this feast day in the Slovak calendar :)
 
In regards to i10n, I am sorry for the confusion: I am planning to merge the new rewrite of `romcal` in ES6 in the `romcal-es6` branch to `master`. Therefore, if you make a PR from this branch, I'll merge it and then merge everything back to master. Also, thanks for taking the time to contribute the localisations :)

**Comment by tukusejssirs** [comment]
As far as I know it correctly, the Divine Office (chiefly used in the British Isles and Former British Colonies) and the Liturgy of Hours (generally used in the Americas and to some extent generally in all English-speaking countries). It contains the texts of the ‘prayer of the Church.’ It is also called breviary.

But the book I meant, looks like [this](https://www.zachej.sk/produkt/26573/direktorium-2018/) from outside and like [this](https://www.liturgie.ch/images/liturgie/liturgiepraxis/direktorium/pdf/Direktorium2018_2018_03_Maerz.pdf) from inside (though this is a German version of it, for I haven’t found any image on the Internet in Slovak).

Yes, I am aware of solemnity/feast/memeria/optional memoria/feria priority order. ;)

If you’d like to incorporate any ‘standard’ votive Holy Masses, you can start with ferial Saturdays (Holy Mass to honour Our Lady / Blessed Virgin Mary) and ferial first Fridays (Holy Mass to honour the Most Holy Sacred Heart of Jesus). :) … But on the other hand, I don’t think that the votive Holy Masses incorporation is important, but still—it may be useful in some cases. :)

I am not sure, when I finish translation of the strings (I have many other duties and 805 lines to translate), but I do my best. I checked that both files (in  both `master` and `romcal-es6` branches) are the same except for the name. :)

I’d like to list all Holy Days of Obligation in Slovakia:
1. 1st Jan: Mary, Mother of God
2. 6th Feb: Epiphany
3. Ascension of the Lord (40th day after Easter)
4. Corpus Christi (2nd Thursday after the Pentecost Sunday)
5. 29th Jun: Peter and Paul, Apostles
6. 15th Aug: Assumption of the Blessed Virgin Mary
7. 1st Nov: All Saints
8. 8th Dec: Immaculate Conception
9. 25th Dec: Nativity of the Lord (Christmas)

Note: St Joseph, Husband of Mary is dispensed in Slovakia (that is, we don’t celebrate it as a Holy Day of Obligation), but still it is a Solemnity.assumption

Btw, while I was searching for the English names of the Holy Days of Obligation, I probably found more duplicates—see for yourself.
$ cat en.json | grep -i assumption
    "assumption": "The Assumption of the Blessed Virgin Mary",
    "assumption": "The Assumption of the Blessed Virgin Mary",
    "assumptionOfTheBlessedVirginMaryPrincipalPatronessOfFrance": "Assumption of The Blessed Virgin Mary, Principal Patroness of France",

$ cat en.json | grep -i "all saints"
    "allSaints": "All Saints",
    "allSaints": "All Saints",
    "allSaintsOfIreland": "All Saints of Ireland",
    "allSaintsOfWales": "All Saints of Wales",

Note: Although we usually we don’t celebrate the Assumption of the Blessed Virgin Mary, Principal Patroness of France, Immaculate Conception of The Blessed Virgin Mary, Principal Patroness Of The Philippines, All Saints of Ireland and All Saints of Wales (and probably others), they _might_ be included in Slovakian calendar, but only as *Optional Memorials*. However, probably the best way would be to exclude them from `sk.json` (and probably even from en.json) for a possible confusion.

**Comment by tukusejssirs** [comment]
I probably should translate the `general.json` (to omit national Feasts from other national calendars), but it should be created first. :)

**Comment by pejulian** [comment]
Hi @tukusejssirs 

Thank you for your findings!
I have noted the changes to be made for the duplicates found.

However, I was and still am a little confused with the feast of All Saints in Ireland and Wales:

In the General Roman Calendar for [Ireland](https://en.wikipedia.org/wiki/General_Roman_Calendar#Ireland) and [Wales](https://en.wikipedia.org/wiki/General_Roman_Calendar#Wales), there are entries for All Saints on the 6th and 8th November in the respective national calendars of these nations.
 
I am unsure if these are _separate_ feasts in addition to the Solemnity of All Saints on 1st November _or_ they actually refer to the Solemnity itself that is moved to a different date. For now, I have left these celebrations intact until I can get more information about them.

**Comment by tukusejssirs** [comment]
After a quick search, I found [this](http://pilgrimsprogresspddm.blogspot.sk/2012/11/6th-november-feast-of-all-saints-of.html?m=1) website, which says:

‘The scope of this feast, while it includes canonised saints, is wider. It also includes those who had a reputation for holiness and whose causes for canonisation have not yet been completed, such as Blessed Thaddeus MacCarthy (1455-92), the seventeen Irish martyrs of the 16th and 17th centuries, Blessed Edmund Ignatius Rice (1762-1844), Blessed Columba Marmion (1858-1923) and the Servant of God Matt Talbot (1856-1925) and people like Legion of Mary envoys Edel Quinn and Alfie Lamb, whose causes have already been introduced. But it also includes those whose lives of sanctity were known only to their families, friends or members of their parish diocese or religious community.

‘Interestingly enough, we are not the only country to have a ‘second’ All Saint’s Day. In the Russian Orthodox Church, the third Sunday after Pentecost, is also the Feast of All Saints of Britain and Ireland. This feast was established by decree of the Russian Orthodox Church on August 21, 2007. This same Sunday is also the commemoration of All Saint of Novgorod, All Saints of Volgoda, All Saints of Belarus, All Saints of Pskov, and All Saints of St Petersburg.’

Therefore, I understand that these feasts of All Saints of {Ireland | Scotland | etc} are additional feasts to the general All Saints one (1st Nov) and are only included in the particular national calendar. I think, on this we agree. :)

But now I think that the strings (`locales/*json` files) might/should contain all feasfs from all (General Roman and national) calendar, as these are only localisations and have nothing to do with caledars themselves. What I mean is that I could (if I want) generate any national calendar in any language (e.g. Irish national calendar in Slovakian language)—of course, that would require a bit of work with tranlation and you might not agree with me.

**Comment by tukusejssirs** [comment]
I just re-read the comments from this issue and found out that I mislead you.

5th July is a Solemnity in both Slovakia and Czech Republic, not a Feast.

I apologise for this.

feast = sviatok (sk)
solemnity = slávnosť (sk)

And as you can see [here](https://lc.kbs.sk/?den=20170705), beneath the Solemnity name (Svätých Cyrila a Metoda, slovanských vierozvestov), there’s ‘slávnosť’ in brackets.

**Comment by tukusejssirs** [comment]
Excerpt from `Readme.md` from `romcal-es6` branch:

‘For the national calendars of the Czech Republic and Slovakia (country: czechRepublic and slovakia), an additional flag can be passed:

    ‘saintsCyrilMonkAndMethodiusBishopOnFeb14: If true, changes the feast of Saints Cyril and Methodius to fall on 14th February instead of the 5th of July. Defaults to false.’

We never have a fwast nor solemnity on 14th Feb. There might be a mere _votive_ Holy Mass, but on 5th July there is **_always_** a Solemnity. :)

So there might be a flag like this, but only to add votive Holy Mass on 14th Feb ... but still it might be included as a ‘standard’ votive Holy Mass and that would eliminate the need of the flag. :)

**Comment by tukusejssirs** [comment]
As for the translation/l10n/i18n, is there any particular need of the blank line at the end of the locale files?

**Comment by tukusejssirs** [comment]
Found another duplicate:

$ cat skSk.mjs | grep -in anselm
111:    "saintAnselmOfCanterburyBishopAndDoctorOfTheChurch": "Saint Anselm of Canterbury, Bishop and Doctor of the Church",
392:    "saintAnselmOfCanterburyBishopAndDoctorOfTheChurch": "Saint Anselm of Canterbury, Bishop and Doctor of the Church",

**Comment by tukusejssirs** [comment]
Still, I found another which I somehow missed:

$ cat en.mjs | grep -in matthi
123:    "saintMatthiasTheApostle": "Saint Matthias the Apostle",
450:    "saintMatthiasApostle": "Saint Matthias, Apostle",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in alexandr
143:    "saintCyrilOfAlexandriaBishopAndDoctor": "Saint Cyril of Alexandria, Bishop and Doctor",
236:    "saintCatherineOfAlexandriaVirginAndMartyr": "Saint Catherine of Alexandria, Virgin and Martyr",
497:    "saintCyrilOfAlexandriaBishopAndDoctor": "Saint Cyril of Alexandria, Bishop and Doctor",
616:    "saintCatherineOfAlexandriaVirginAndMartyr": "Saint Catherine of Alexandria, Virgin and Martyr",

**Comment by tukusejssirs** [comment]
I would sugget to sort the key alphabetically within `general`, `national` and `celebrations`, but first (I think) the keys must be unified in naming. I propose to remove words like `Saint`, `Saints` and `Blessed` (and probably even conjuctions and `priest`, `deacon`, `religious`, `doctor`, etc) from the keys names. That way it might be more human-readable and less duplicates would occur.

**Comment by tukusejssirs** [comment]
And some of the key names and values (like `saintLadislaus`) have nothing like `bishop` or `virgin`, but simply `{Saint | Blessed} {Name}`.

**Comment by tukusejssirs** [comment]
In (English/General) calendar, on 5th July, Saint Anthony Zaccaria, Priest is celebrated, but in Slovakia it is moved to 7th July

**Comment by tukusejssirs** [comment]
cat en.mjs | grep -in saintApollinaris
156:    "saintApollinaris": "Saint Apollinaris",
669:    "saintApollinaris": "Saint Apollinaris",

**Comment by tukusejssirs** [comment]
These are duplicates, too, at least in Slovakian calendar. The first is in the Slovakian calendar (the three priests are venerated together as (Saints) Three Martyrs of Košice.

"saintsMarkoKrizinMelicharGrodeckiAndStephenPongracPriestsAndMartyrs": "Saints Marko Krizin, Melichar Grodecki and Stephen Pongrac, Priests and Martyrs",

"saintMarkoKrizinPriestAndMartyr": "Saint Marko Krizin, Priest and Martyr",
"saintMelchiorGrodzieckiPriestAndMartyr": "Saint Melchior Grodziecki, Priest and Martyr",

Note 1: As I am not a native English speaker, I am not sure where to put the word ‘Saints’, but in Slovakian language it is at the beginning, because the Three Martyrs are considered as a group noun).

Note 2: As for the names, none of them was Slovak, but the names where slovakised. Here is the table of the slovakised, hungarised/Hungarian and original form of names. Note that Hungarian names start with last names and Christian names are placed on the second place.

| Slovakian form | Original form | Original form language | Hungarian form |
|---|---|---|---|
| Marek Križin | Marko Stjepan Križin or Križevčanin | Croatia | Kőrösi Márk |
| Melichar Grodziecký | Melchior Grodziecki | Poland | Grodecz Menyhért |
| Štefan Pongrácz | István Pongrácz | Hungary | Pongrácz István |

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in birthOfTheBlessedVirginMary
190:    "birthOfTheBlessedVirginMary": "Birth of the Blessed Virgin Mary",
608:    "birthOfTheBlessedVirginMary": "Birth of the Blessed Virgin Mary",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -n GallAb
479:    "saintGallAbbot": "Saint Gall, Abbot",
580:    "saintGallAbbotAndMissionary": "Saint Gall, Abbot and Missionary",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in magall
126:    "saintChristopherMagallanesAndCompanionsMartyrs": "Saint Christopher Magallanes and Companions, Martyrs",
619:    "saintCristobalMagallanesAndCompanionsMartyrs": "Saint Cristobal Magallanes and Companions, Martyrs",

**Comment by pejulian** [comment]
@tukusejssirs 

The duplicates that you have outlined via `grep` have mostly been resolved and commited.

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in arbel
160:    "saintSharbelMakhlufPriestAndHermit": "Saint Sharbel Makhlūf, Priest and Hermit",
264:    "saintSharbelMakhlufPriestAndHermit": "Saint Sharbel Makhlūf, Priest and Hermit",
591:    "saintCharbelPriest": "Saint Charbel, Priest",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in saintTuribiusOfMogrovejoBishop
104:    "saintTuribiusOfMogrovejoBishop": "Saint Turibius of Mogrovejo, Bishop",
257:    "saintTuribiusOfMogrovejoBishop": "Saint Turibius of Mogrovejo, Bishop",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in isidoreof
106:    "saintIsidoreOfSevilleBishopAndDoctorOfTheChurch": "Saint Isidore of Seville, Bishop and Doctor of the Church",
737:    "saintIsidoreOfSevilleBishopAndDoctorOfTheChurch": "Saint Isidore of Seville, Bishop and Doctor of The Church",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in saintsAugustineZhaoRongPriestAndCompanionsMartyrs
150:    "saintsAugustineZhaoRongPriestAndCompanionsMartyrs": "Saints Augustine Zhao Rong, Priest, and Companions, Martyrs",
262:    "saintsAugustineZhaoRongPriestAndCompanionsMartyrs": "Saints Augustine Zhao Rong, Priest, and Companions, Martyrs",

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in ourLadyOfMountCarmel
155:    "ourLadyOfMountCarmel": "Our Lady of Mount Carmel",
263:    "ourLadyOfMountCarmel": "Our Lady of Mount Carmel",
357:    "ourLadyOfMountCarmelMotherAndQueenOfChile": "Our Lady of Mount Carmel, Mother and Queen of Chile",

$ cat en.mjs | grep -in saintRoseOfLima
183:    "saintRoseOfLimaVirgin": "Saint Rose of Lima, Virgin",
268:    "saintRoseOfLima": "Saint Rose of Lima",
296:    "saintRoseOfLimaVirgin": "Saint Rose of Lima, Virgin",

$ cat en.mjs | grep -in saintsRoqueGonzalezAlf
272:    "saintsRoqueGonzalezAlfonsoRodriguezAndJuanDelCastilloPriestsAndMartyrs": "Saints Roque Gonzalez, Alfonso Rodriguez, and Juan del Castillo, Priests and Martyrs",
304:    "saintsRoqueGonzalezAlfonsoRodriguezOlmedoAndJuanDelCastilloPriestsAndMartyrs": "Saints Roque Gonzalez, Alfonso Rodriguez Olmedo, and Juan del Castillo, Priests and Martyrs",

$ grep -in -A2 saintElizabethOfHungary ../calendars/*
../calendars/argentina.mjs:166:      "key": "saintElizabethOfHungary",
../calendars/argentina.mjs-167-      "type": Types[5],
../calendars/argentina.mjs-168-      "moment": moment.utc({ year: year, month: 10, day: 18 }),
--
../calendars/england.mjs:405:      "key": "saintHildaAbbessOrSaintHughOfLincolnBishopOrSaintElizabethOfHungary",
../calendars/england.mjs-406-      "type": Types[6],
../calendars/england.mjs-407-      "moment": moment.utc({ year: year, month: 10, day: 17 }),
--
../calendars/general.mjs:1562:      "key": "saintElizabethOfHungaryReligious",
../calendars/general.mjs-1563-      "type": Types[5],
../calendars/general.mjs-1564-      "moment": moment.utc({ year: year, month: 10, day: 17 }),
--
../calendars/germany.mjs:463:      "key": "saintElizabethOfHungaryReligious",
../calendars/germany.mjs-464-      "type": Types[5],
../calendars/germany.mjs-465-      "moment": moment.utc({ year: year, month: 10, day: 19 }),

**Comment by tukusejssirs** [comment]
$ cat en.mjs | grep -in ourLadyOfGuadalupe
244:    "ourLadyOfGuadalupe": "Our Lady of Guadalupe",
274:    "ourLadyOfGuadalupe": "Our Lady of Guadalupe",

$ cat en.mjs | grep -in saintPatrickBishop
102:    "saintPatrickBishop": "Saint Patrick, Bishop",
277:    "saintPatrickBishop": "Saint Patrick, Bishop",

**Comment by pejulian** [comment]
```
$ cat en.mjs | grep -in "barbara\|damasc"
239: "saintJohnDamascenePriestAndDoctor": "Saint John Damascene, Priest and Doctor",
486: "saintBarbaraVirginAndMartyrOrBlessedAdolphKolpingPriest": "Saint Barbara, Virgin and Martyr Or Blessed Adolph Kolping, Priest",
499: "saintJohnDamasceneOrSaintBarbara": "Saint John Damascene Or Saint Barbara",
589: "saintBarbaraVirginAndMartyr": "Saint Barbara, Virgin and Martyr",

Note: Probably saintBarbaraVirginAndMartyrOrBlessedAdolphKolpingPriest key is not a duplicate, but as I am unsure, I kept it in the output.
```

Since all of these occurrences are in national calendars, they would not represent a problem since they will be unique to the country's calendar output. Therefore, I think it safe to leave the keys as they are now.

**Comment by tukusejssirs** [comment]
Possible duplicates are `ourLadyOfTheValley` and `birthOfTheBlessedVirginMary`. The `ourLadyOfTheValley` is venerated in Argentina and Venezuela (the is omitted as there is no calendar for Venezuela yet; see [this link](https://en.wikipedia.org/wiki/Virgen_del_Valle)). There is even Our Lady of Coromoto venerated in Venezuela (see [this](https://en.wikipedia.org/wiki/Our_Lady_of_Coromoto)).

Anyways, this is probably a question for those living in those countries.

**Comment by tukusejssirs** [comment]
Another possible duplicate:

$ cat ../locales/en.mjs | grep -n saintPeterChanelPriestAndMartyr
115:    "saintPeterChanelPriestAndMartyrSaintLouisGrignonDeMontfortPriest": "Saint Peter Chanel, Priest and Martyr/Saint Louis Grignon de Montfort, Priest",
270:    "saintPeterChanelPriestAndMartyr": "Saint Peter Chanel, Priest and Martyr",

**Comment by tukusejssirs** [comment]
$ cat enUs.mjs | grep -in damien
278:    "saintFatherDamienMissionary": "Saint Father Damien, Missionary",
750:    "saintDamienDeVeusterPriest": "Saint Damien de Veuster, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in lambert enUs.mjs 
281:    "saintLambertBishopAndMartyr": "Saint Lambert, Bishop and Martyr",
460:    "saintLambertOfMaastrichtBishopAndMartyr": "Saint Lambert of Maastricht, Bishop and Martyr",

**Comment by tukusejssirs** [comment]
Possible duplicates, duplicates and (possible) joined celebrations:

$ grep -in Miki enUs.mjs
89:    "saintPaulMikiAndCompanionsMartyrs": "Saint Paul Miki and Companions, Martyrs",
284:    "saintsFelipeDeJesusPaulMikiAndCompanionsMartyrs": "Saints Felipe de Jesus, Paul Miki and Companions, Martyrs",
574:    "saintPaulMikiAndCompanionsMartyrs": "Saint Paul Miki and Companions, Martyrs",
621:    "saintsPedroBautistaPaulMikiAndCompanionsMartyrs": "Saints Pedro Bautista, Paul Miki and Companions, Martyrs",

**Comment by tukusejssirs** [comment]
Duplicates and joined feasts:

$ grep -in saintCamillusDeLellisPriest enUs.mjs
153:    "saintCamillusDeLellisPriest": "Saint Camillus de Lellis, Priest",
287:    "saintCamillusDeLellisPriest": "Saint Camillus de Lellis, Priest",
347:    "saintCamillusDeLellisPriestOrSaintHenry": "Saint Camillus de Lellis, Priest, Or Saint Henry",

$ grep -in saintPeterClaverPriest enUs.mjs 
191:    "saintPeterClaverPriest": "Saint Peter Claver, Priest",
288:    "saintPeterClaverPriest": "Saint Peter Claver, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in saintAnthonyMaryClaretBishop enUs.mjs 
219:    "saintAnthonyMaryClaretBishop": "Saint Anthony Mary Claret, Bishop",
293:    "saintAnthonyMaryClaretBishop": "Saint Anthony Mary Claret, Bishop",

$ grep -in saintMartinDePorresReligious enUs.mjs 
222:    "saintMartinDePorresReligious": "Saint Martin de Porres, Religious",
294:    "saintMartinDePorresReligious": "Saint Martin de Porres, Religious",

**Comment by tukusejssirs** [comment]
$ grep -in saintScholasticaVirgin enUs.mjs 
91:    "saintScholasticaVirgin": "Saint Scholastica, Virgin",
297:    "saintScholasticaVirgin": "Saint Scholastica, Virgin",

$ grep -in saintCatherineOfSienaVirginAndDoctorOfTheChurch enUs.mjs
116:    "saintCatherineOfSienaVirginAndDoctorOfTheChurch": "Saint Catherine of Siena, Virgin and Doctor of The Church, Patron of Europe",
300:    "saintCatherineOfSienaVirginAndDoctorOfTheChurch": "Saint Catherine of Siena, Virgin and Doctor of The Church, Patron of Europe",

**Comment by tukusejssirs** [comment]
$ grep -in saintBirgittaReligious ../locales/enUs.mjs
159:    "saintBirgittaReligious": "Saint Birgitta, Religious (Saint Briget of Sweden)",
306:    "saintBirgittaReligious": "Saint Birgitta, Religious",

**Comment by tukusejssirs** [comment]
I believe you omitted St Elisabeth of Hungary duplicates from [one of my previous comments](https://github.com/pejulian/romcal/issues/10#issuecomment-372153186):

```bash
$ grep -in saintElizabethOfHungary enUs.mjs
267:    "saintElizabethOfHungary": "Saint Elizabeth of Hungary",
399:    "saintElizabethOfHungary": "Saint Hilda, Abbess/Saint Hugh of Lincoln, Bishop/Saint Elizabeth of Hungary",
```

**Comment by tukusejssirs** [comment]
$ grep -in saintRaymondOfPenyafortPriest enUs.mjs
75:    "saintRaymondOfPenyafortPriest": "Saint Raymond of Penyafort, Priest",
311:    "saintRaymondOfPenyafortPriest": "Saint Raymond of Penyafort, Priest",

**Comment by tukusejssirs** [comment]
Saint Joseph, Husband of the Blessed Virgin Mary is Solemnity in the General Roman Calendar ([src](https://en.wikipedia.org/wiki/General_Roman_Calendar#March)), therefore:

(1) it should be added to `general.mjs` calendar;
(2) the following are duplicates (because in Canada, St Joseph, Spouse of the Blessed Virgin Mary, Principal Patron of Canada is celebrated on [19th March](https://www.catholicculture.org/culture/liturgicalyear/calendar/day.cfm?date=2018-03-19)):
```bash
$ grep -in "husband\|spouse"  enUs.mjs ../calendars/*
enUs.mjs:49:    "josephHusbandOfMary": "Joseph, Husband of Mary",
enUs.mjs:313:    "saintJosephSpouseOfTheBlessedVirginMaryPrincipalPatronOfCanada": "Saint Joseph, Spouse of The Blessed Virgin Mary, Principal Patron of Canada",
../calendars/canada.mjs:37:      "key": "saintJosephSpouseOfTheBlessedVirginMaryPrincipalPatronOfCanada",
```

(3) create `enCa.mjs` locale with the following content:
```bash
export default {
  "general": {
    "josephHusbandOfMary": "Saint Joseph, Spouse of The Blessed Virgin Mary, Principal Patron of Canada",
  }
};
```

**Comment by tukusejssirs** [comment]
$ grep -in LouisGrignon enUs.mjs 
115:    "saintPeterChanelPriestAndMartyrSaintLouisGrignonDeMontfortPriest": "Saint Peter Chanel, Priest and Martyr/Saint Louis Grignon de Montfort, Priest",
597:    "saintLouisGrignonDeMontfortPriest": "Saint Louis Grignon de Montfort, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in joachim enUs.mjs 
162:    "saintsJoachimAndAnne": "Saints Joachim and Anne",
324:    "saintAnnePatronOfQuebecAndSaintJoachimParentsOfTheBlessedVirginMary": "Saint Anne, Patron of Quebec, and Saint Joachim, Parents of The Blessed Virgin Mary",

**Comment by tukusejssirs** [comment]
$ grep -in ourLadyOfLourdes enUs.mjs
92:    "ourLadyOfLourdes": "Our Lady of Lourdes",
334:    "ourLadyOfLourdes": "Our Lady of Lourdes",

**Comment by tukusejssirs** [comment]
$ grep -in exaltation enUs.mjs
67:    "theExaltationOfTheHolyCross": "The Exaltation of the Holy Cross",
335:    "exaltationOfTheHolyCross": "Exaltation of The Holy Cross",

**Comment by tukusejssirs** [comment]
$ grep -in saintsPhilipAndJamesApostles enUs.mjs 
120:    "saintsPhilipAndJamesApostles": "Saints Philip and James, Apostles",
336:    "saintsPhilipAndJamesApostles": "Saints Philip and James, Apostles",

**Comment by tukusejssirs** [comment]
$ grep -in jesusChristTheEternalHighPriest enUs.mjs 
289:    "ourLordJesusChristTheEternalHighPriest": "Our Lord Jesus Christ, The Eternal High Priest",
341:    "jesusChristTheEternalHighPriest": "Jesus Christ, The Eternal High Priest",

**Comment by tukusejssirs** [comment]
$ grep -in saintTeresaBenedictaOfTheCrossEdithSteinVirginAndMartyr enUs.mjs
172:    "saintTeresaBenedictaOfTheCrossEdithSteinVirginAndMartyr": "Saint Teresa Benedicta of The Cross (Edith Stein), Virgin and Martyr, Patron of Europe",
343:    "saintTeresaBenedictaOfTheCrossEdithSteinVirginAndMartyr": "Saint Teresa Benedicta of The Cross (Edith Stein), Virgin and Martyr, Patron of Europe",

**Comment by tukusejssirs** [comment]
Here is a list of duplicate keys, as well as keys with celebrations that should be separated (issue #12). Also note that in `lithuania.mjs`, the celebrate date and type is the same as in `general.mjs`.

```bash $ grep -inA2 saintAdalbertBishopAndMartyr enUs.mjs ../calendars/{general,england,lithuania}.mjs
enUs.mjs:112:    "saintGeorgeMartyrSaintAdalbertBishopAndMartyr": "Saint George, Martyr/Saint Adalbert, Bishop and Martyr",
enUs.mjs-113-    "saintFidelisOfSigmaringenPriestAndMartyr": "Saint Fidelis of Sigmaringen, Priest and Martyr",
enUs.mjs-114-    "saintMarkTheEvangelist": "Saint Mark the Evangelist",
--
enUs.mjs:346:    "saintAdalbertBishopAndMartyr": "Saint Wojciech (Adalbert of Prague), Bishop and Martyr",
enUs.mjs-347-    "saintSigismundMartyr": "Saint Sigismund, Martyr",
enUs.mjs-348-    "saintJohnNepomucenePriestAndMartyr": "Saint John Nepomucene, Priest and Martyr",
--
enUs.mjs:371:    "saintAdalbertBishopAndMartyrSaintFidelisOfSigmaringenPriestAndMartyr": "Saint Wojciech (Adalbert of Prague), Bishop and Martyr/Saint Fidelis of Sigmaringen, Priest and Martyr",
enUs.mjs-372-    "theEnglishMartyrs": "The English Martyrs",
enUs.mjs-373-    "stDunstanArchbishopOfCanterbury": "St Dunstan, Archbishop of Canterbury",
--
../calendars/general.mjs:397:      "key": "saintGeorgeMartyrSaintAdalbertBishopAndMartyr",
../calendars/general.mjs-398-      "type": Types[6],
../calendars/general.mjs-399-      "moment": moment.utc({ year: year, month: 3, day: 23 }),
--
../calendars/england.mjs:90:      "key": "saintAdalbertBishopAndMartyrSaintFidelisOfSigmaringenPriestAndMartyr",
../calendars/england.mjs-91-      "type": Types[6],
../calendars/england.mjs-92-      "moment": moment.utc({ year: year, month: 4, day: 24 }),
--
../calendars/lithuania.mjs:47:      "key": "saintAdalbertBishopAndMartyr",
../calendars/lithuania.mjs-48-      "type": Types[6],
../calendars/lithuania.mjs-49-      "moment": moment.utc({ year: year, month: 3, day: 23 }),
```

**Comment by tukusejssirs** [comment]
I’ve just update one of my previous [comment](https://github.com/pejulian/romcal/issues/10#issuecomment-373978571), but then I realised you solved that duplicate and these are newly emerged duplicates, so I’ve copied it here:

$ grep -in "birgitta\|brigitta\|bridget" enUs.mjs 
159:    "saintBirgittaReligious": "Saint Birgitta, Religious (Saint Briget of Sweden)",
297:    "saintBirgittaReligious": "Saint Birgitta, Religious",
356:    "saintBridgetOfSweedenReligious": "Saint Bridget of Sweeden, Religious, Patron of Europe",
410:    "saintBrigittaReligious": "Saint Brigitta, Religious",

**Comment by tukusejssirs** [comment]
$ grep -in wolfgang enUs.mjs 
364:    "saintWolfgangBishop": "Saint Wolfgang, Bishop",
458:    "saintWolfgangOfRegensburgBishop": "Saint Wolfgang of Regensburg, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in hedwigReligious enUs.mjs 
214:    "saintHedwigReligiousSaintMargaretMaryAlacoqueVirgin": "Saint Hedwig, Religious/Saint Margaret Mary Alacoque, Virgin",
330:    "saintHedwigReligiousOrSaintMargaretMaryAlacoqueVirgin": "Saint Hedwig, Religious Or Saint Margaret Mary Alacoque, Virgin",

**Comment by tukusejssirs** [comment]
$ grep -in saintPiusVPope enUs.mjs 
117:    "saintPiusVPope": "Saint Pius V, Pope",
578:    "saintPiusVPope": "Saint Pius V, Pope",

**Comment by tukusejssirs** [comment]
Possible duplicate (needs more info from someone who knows better the liturgy in England, and Wales):
```bash
$ grep -in "theEnglishMartyrs\|theSixWelsh" enUs.mjs 
369:    "theEnglishMartyrs": "The English Martyrs",
748:    "theSixWelshMartyrsAndCompanions": "The Six Welsh Martyrs and Companions",
```

Links:
- [universalis](http://universalis.com/20180504/today.htm),
- [wiki (40 Martyrs)](https://en.wikipedia.org/wiki/Forty_Martyrs_of_England_and_Wales),
- [liturgyoffice (4th May)](http://www.liturgyoffice.org.uk/Calendar/National/England2.shtml#May4),
- [liturgyoffice (25th Oct)](http://www.liturgyoffice.org.uk/Calendar/National/Wales2.shtml#Oct25).

What I suggest (if the celebrations are the same, but only for partial number of saints) is to create only one key (e.g. `fourtyMartyrsOfEnglandAndWales`), but create two subkeys (e.g. `fourtyMartyrsOfEnglandAndWales.england` and `fourtyMartyrsOfEnglandAndWales.wales`) to change the name of the celebration (see issue #24).

In the end, this is probably not important at all, but still I wanted to write it down.

**Comment by tukusejssirs** [comment]
$ grep -in AugustineOfCanterbury enUs.mjs 
131:    "saintAugustineOfCanterburyBishop": "Saint Augustine of Canterbury, Bishop",
372:    "saintAugustineOfCanterburyBishop": "Saint Augustine of Canterbury, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in saintBonifaceBishopAndMartyr  enUs.mjs 
136:    "saintBonifaceBishopAndMartyr": "Saint Boniface, Bishop and Martyr",
373:    "saintBonifaceBishopAndMartyr": "Saint Boniface, Bishop and Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in saintGregoryTheGreatPopeAndDoctor enUs.mjs 
190:    "saintGregoryTheGreatPopeAndDoctor": "Saint Gregory the Great, Pope and Doctor of the Church",
383:    "saintGregoryTheGreatPopeAndDoctor": "Saint Gregory The Great, Pope and Doctor",

**Comment by tukusejssirs** [comment]
$ grep -n peterAndPaulApostles enUs.mjs 
59:    "peterAndPaulApostles": "Saints Peter and Paul, Apostles",
389:    "peterAndPaulApostles": "Saints Peter and Paul, Apostles",

**Comment by tukusejssirs** [comment]
$ grep -in "saintAndrewApostle\|saintAndrewTheApostle" enUs.mjs 
239:    "saintAndrewTheApostle": "Saint Andrew the Apostle",
397:    "saintAndrewApostle": "Saint Andrew, Apostle",
693:    "saintAndrewTheApostlePatronOfRussia": "Saint Andrew The Apostle, Patron of Russia",

**Comment by tukusejssirs** [comment]
$ grep -in saintThomasBecketBishopAndMartyr enUs.mjs 
254:    "saintThomasBecketBishopAndMartyr": "Saint Thomas Becket, Bishop and Martyr",
398:    "saintThomasBecketBishopAndMartyr": "Saint Thomas Becket, Bishop and Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in saintHenryBishopAndMartyr enUs.mjs 
153:    "saintHenryBishopAndMartyr": "Saint Henry, Bishop and Martyr",
399:    "saintHenryBishopAndMartyr": "Saint Henry, Bishop and Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in saintPeterCanisiusPriestAndDoctor enUs.mjs 
249:    "saintPeterCanisiusPriestAndDoctor": "Saint Peter Canisius, Priest and Doctor",
430:    "saintPeterCanisiusPriestAndDoctor": "Saint Peter Canisius, Priest and Doctor",

**Comment by tukusejssirs** [comment]
$ grep -in saintCyrilOfJerusalemBishopAndDoctor enUs.mjs 
103:    "saintCyrilOfJerusalemBishopAndDoctor": "Saint Cyril of Jerusalem, Bishop and Doctor",
465:    "saintCyrilOfJerusalemBishopAndDoctor": "Saint Cyril of Jerusalem, Bishop and Doctor",

**Comment by tukusejssirs** [comment]
$ grep -in ourLadyOfFatima enUs.mjs 
122:    "ourLadyOfFatima": "Our Lady of Fatima",
467:    "ourLadyOfFatima": "Our Lady of Fatima",

**Comment by tukusejssirs** [comment]
$ grep -in saintsCosmasAndDamian enUs.mjs 
202:    "saintsCosmasAndDamianMartyrs": "Saints Cosmas and Damian, Martyrs",
471:    "saintsCosmasAndDamian": "Saints Cosmas and Damian",

**Comment by tukusejssirs** [comment]
$ grep -in presentationOfTheBlessedVirginMary enUs.mjs 
234:    "presentationOfTheBlessedVirginMary": "Presentation of the Blessed Virgin Mary",
474:    "presentationOfTheBlessedVirginMary": "Presentation of The Blessed Virgin Mary",

**Comment by tukusejssirs** [comment]
@pejulian, I’ve just found out that the `saintJohnDamasceneOrSaintBarbara` is not in any national calendar in `src/calendars/*` using `grep -in saintJohnDamasceneOrSaintBarbara src/calendars/*` (no output).

Therefore, this founding contradicts to one of your previous [comments](https://github.com/pejulian/romcal/issues/10#issuecomment-372179842) and it should be deleted from `src/locales/*`.

**Comment by tukusejssirs** [comment]
$ grep -in '"saintNicholas"\|saintNicholasBishop' enUs.mjs 
242:    "saintNicholasBishop": "Saint Nicholas, Bishop",
476:    "saintNicholas": "Saint Nicholas",
564:    "saintNicholasBishop": "Saint Nicholas, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in saintStephenOfHungary enUs.mjs 
179:    "saintStephenOfHungary": "Saint Stephen of Hungary",
488:    "saintStephenOfHungary": "Saint Stephen of Hungary",

**Comment by tukusejssirs** [comment]
$ grep -in saintThomasTheApostle enUs.mjs 
147:    "saintThomasTheApostle": "Saint Thomas the Apostle",
501:    "saintThomasTheApostle": "Saint Thomas The Apostle",

**Comment by tukusejssirs** [comment]
$ grep -in saintFrancisXavierPriest enUs.mjs 
240:    "saintFrancisXavierPriest": "Saint Francis Xavier, Priest",
503:    "saintFrancisXavierPriest": "Saint Francis Xavier, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in saintCasimir enUs.mjs 
98:    "saintCasimir": "Saint Casimir",
569:    "saintCasimir": "Saint Casimir",

**Comment by tukusejssirs** [comment]
```bash
$ grep -inA2 pioOfPietr ../calendars/*
../calendars/general.mjs:1280:      "key": "saintPioOfPietrelcinaPadrePioPriest",
../calendars/general.mjs-1281-      "type": Types[5],
../calendars/general.mjs-1282-      "moment": moment.utc({ year: year, month: 8, day: 23 }),
--
../calendars/ireland.mjs:368:      "key": "saintPioOfPietralcina",
../calendars/ireland.mjs-369-      "type": Types[5],
../calendars/ireland.mjs-370-      "moment": moment.utc({ year: year, month: 8, day: 23 }),
```

Note that in `saintPioOfPiotralcina` has a typo, which you have copied from the [General Roman Calendar](https://en.wikipedia.org/wiki/General_Roman_Calendar#Ireland).

**Comment by tukusejssirs** [comment]
Probably by mistake (or ‘mis-stroke’), you duplicated the `ourLadyOfLujanPatronessOfArgentina`:
```bash
$ grep -in ourLadyOfLujanPatronessOfArgentina enUs.mjs 
259:    "ourLadyOfLujanPatronessOfArgentina": "Our Lady of Luján, Patroness of Argentina",
260:    "ourLadyOfLujanPatronessOfArgentina": "Our Lady of Luján, Patroness of Argentina",
```

**Comment by tukusejssirs** [comment]
$ grep -in saintStanislausBishopAndMartyr enUs.mjs 
109:    "saintStanislausBishopAndMartyr": "Saint Stanislaus, Bishop and Martyr",
615:    "saintStanislausBishopAndMartyr": "Saint Stanislaus, Bishop and Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in saintMariaGorettiVirginAndMartyr enUs.mjs 
150:    "saintMariaGorettiVirginAndMartyr": "Saint Maria Goretti, Virgin and Martyr",
625:    "saintMariaGorettiVirginAndMartyr": "Saint Maria Goretti, Virgin and Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in "Ruiz" enUs.mjs 
204:    "saintWenceslausMartyrSaintsLawrenceRuizAndCompanionsMartyrs": "Saint Wenceslaus, Martyr/Saints Lawrence Ruiz and Companions, Martyrs",
607:    "saintLorenzoRuizAndCompanionsMartyrs": "Saint Lorenzo Ruiz and Companions, Martyrs",
639:    "saintsLawrenceRuizAndCompanionsMartyrs": "Saints Lawrence Ruiz and Companions, Martyrs",

Note: Lawrence = Lorenzo.

**Comment by tukusejssirs** [comment]
$ grep -in saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest enUs.mjs 
213:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest": "Saint Denis and Companions Martyrs/Saint John Leonardi, Priest",
635:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest": "Saint Denis and Companions Martyrs/Saint John Leonardi, Priest",
669:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriestBlessedJohnNewmanBishop": "Saint Denis and Companions, Martyrs/Saint John Leonardi, Priest/Blessed John Newman, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in saintJohnOfKantyPriest enUs.mjs 
250:    "saintJohnOfKantyPriest": "Saint John of Kanty, Priest",
647:    "saintJohnOfKantyPriest": "Saint John of Kanty, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in JohnOfGod enUs.mjs 
100:    "saintJohnOfGodReligious": "Saint John of God, Religious",
660:    "saintJohnOfGodPriest": "Saint John of God, Priest",

**Comment by tukusejssirs** [comment]
$ grep -in saintAndrewTheApostle enUs.mjs 
239:    "saintAndrewTheApostle": "Saint Andrew the Apostle",
693:    "saintAndrewTheApostlePatronOfRussia": "Saint Andrew The Apostle, Patron of Russia",
700:    "saintAndrewTheApostle": "Saint Andrew The Apostle",

**Comment by tukusejssirs** [comment]
$ grep -in saintPatrick enUs.mjs 
102:    "saintPatrickBishop": "Saint Patrick, Bishop",
272:    "saintPatrickBishop": "Saint Patrick, Bishop",
696:    "saintPatrick": "Saint Patrick",

**Comment by tukusejssirs** [comment]
We need to deal with these, too (St Columban and St Columba / Colum Cille are two different saints):
```bash
$ grep -inA2 saintColumba ../calendars/*
../calendars/england.mjs:166:      "key": "saintEphraemDeaconDoctorOrSaintColumbaColumCilleAbbot",
../calendars/england.mjs-167-      "type": Types[6],
../calendars/england.mjs-168-      "moment": moment.utc({ year: year, month: 5, day: 9 }),
--
../calendars/general.mjs:1629:      "key": "saintClementIPopeAndMartyrSaintColumbanReligious",
../calendars/general.mjs-1630-      "type": Types[6],
../calendars/general.mjs-1631-      "moment": moment.utc({ year: year, month: 10, day: 23 }),
--
../calendars/ireland.mjs:209:      "key": "saintColumbaAbbotAndMissionary",
../calendars/ireland.mjs-210-      "type": Types[4],
../calendars/ireland.mjs-211-      "moment": moment.utc({ year: year, month: 5, day: 9 }),
--
../calendars/ireland.mjs:453:      "key": "saintColumbanAbbotAndMissionary",
../calendars/ireland.mjs-454-      "type": Types[5],
../calendars/ireland.mjs-455-      "moment": moment.utc({ year: year, month: 10, day: 23 }),
--
../calendars/scotland.mjs:66:      "key": "saintColumba",
../calendars/scotland.mjs-67-      "type": Types[5],
../calendars/scotland.mjs-68-      "moment": moment.utc({ year: year, month: 5, day: 9 }),

$ grep -in saintColumba enUs.mjs 
236:    "saintClementIPopeAndMartyrSaintColumbanReligious": "Saint Clement I, Pope and Martyr/Saint Columban, Religious",
374:    "saintEphraemDeaconDoctorOrSaintColumbaColumCilleAbbot": "Saint Ephraem, Deacon, Doctor or Saint Columba (Colum Cille), Abbot",
528:    "saintColumbaAbbotAndMissionary": "Saint Columba, Abbot and Missionary",
553:    "saintColumbanAbbotAndMissionary": "Saint Columban, Abbot and Missionary",
697:    "saintColumba": "Saint Columba",
```

**Comment by tukusejssirs** [comment]
$ grep -in saintVincentDeaconAndMartyr enUs.mjs 
80:    "saintVincentDeaconAndMartyr": "Saint Vincent, Deacon and Martyr",
706:    "saintVincentDeaconAndMartyr": "Saint Vincent, Deacon and Martyr",
731:    "saintVincentDeaconAndMartyrOrSaintMarianneCopeVirgin": "Saint Vincent, Deacon and Martyr Or Saint Marianne Cope, Virgin",

**Comment by tukusejssirs** [comment]
$ grep -in saintJamesApostle enUs.mjs 
162:    "saintJamesApostle": "Saint James, Apostle",
715:    "saintJamesApostlePatronOfSpain": "Saint James, Apostle, Patron of Spain",

**Comment by tukusejssirs** [comment]
$ grep -in saintTeresaOfJesusVirginAndDoctorOfTheChurch enUs.mjs 
208:    "saintTeresaOfJesusVirginAndDoctorOfTheChurch": "Saint Teresa of Jesus, Virgin and Doctor",
719:    "saintTeresaOfJesusVirginAndDoctorOfTheChurch": "Saint Teresa of Jesus, Virgin and Doctor of The Church",

**Comment by tukusejssirs** [comment]
$ grep -inA2 saintPaulOfTheCrossPriest ../calendars/unitedStates.mjs 
112:      "key": "saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrsSaintPaulOfTheCrossPriest",
113-      "type": Types[5],
114-      "moment": moment.utc({ year: year, month: 9, day: 19 }),
--
125:      "key": "saintPaulOfTheCrossPriest",
126-      "type": Types[6],
127-      "moment": moment.utc({ year: year, month: 9, day: 20 }),

**Comment by tukusejssirs** [comment]
$ grep -in GertrudeOfNivelles enUs.mjs 
278:    "saintGertrudeOfNivellesVirgin": "Saint Gertrude of Nivelles, Virgin",
426:    "saintGertrudeOfNivellesAbbess": "Saint Gertrude of Nivelles, Abbess",

**Comment by tukusejssirs** [comment]
$ grep -in florian enUs.mjs 
431:    "saintFlorianAndHisCompanionsMartyrs": "Saint Florian and his Companions, Martyrs",
479:    "saintFlorianMartyr": "Saint Florian, Martyr",

**Comment by tukusejssirs** [comment]
$ grep -in Wolfgang enUs.mjs 
361:    "saintWolfgangBishop": "Saint Wolfgang, Bishop",
454:    "saintWolfgangOfRegensburgBishop": "Saint Wolfgang of Regensburg, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in leonardi enUs.mjs 
213:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest": "Saint Denis and Companions Martyrs/Saint John Leonardi, Priest",
635:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriest": "Saint Denis and Companions Martyrs/Saint John Leonardi, Priest",
669:    "saintDenisAndCompanionsMartyrsSaintJohnLeonardiPriestBlessedJohnNewmanBishop": "Saint Denis and Companions, Martyrs/Saint John Leonardi, Priest/Blessed John Newman, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in "Zoe*rard" enUs.mjs 
485:    "saintsAndrewZorardAndBenedictHermits": "Saints Andrew Zorard and Benedict, Hermits",
701:    "saintsAndrewZoerardusAndBenedictHermits": "Saints Andrew Zoerardus and Benedict, Hermits",

**Comment by tukusejssirs** [comment]
$ grep -in lindisfa enUs.mjs 
382:    "saintAidanBishopAndTheSaintsOfLindisfarne": "Saint Aidan, Bishop and The Saints of Lindisfarne",
541:    "saintAidanOfLindisfarneBishopAndMissionary": "Saint Aidan of Lindisfarne, Bishop and Missionary",

**Comment by tukusejssirs** [comment]
$ grep -inA1 "immaculateConception " ../lib/Dates.mjs && grep -inA2 immaculateConceptionOfTheBlessedVirginMaryPrincipalPatronessOfThePhilippines ../calendars/*
536:const immaculateConception = y => {
537-  let _date = moment.utc({ year: y, month: 11, day: 8 });
../calendars/philippines.mjs:74:      "key": "immaculateConceptionOfTheBlessedVirginMaryPrincipalPatronessOfThePhilippines",
../calendars/philippines.mjs-75-      "type": Types[0],
../calendars/philippines.mjs-76-      "moment": moment.utc({ year: year, month: 11, day: 8 }),

**Comment by tukusejssirs** [comment]
```$ grep -inA2 saintAnthonyOf.*PriestAndDoctor ../calendars/*
../calendars/general.mjs:663:      "key": "saintAnthonyOfPaduaPriestAndDoctor",
../calendars/general.mjs-664-      "type": Types[5],
../calendars/general.mjs-665-      "moment": moment.utc({ year: year, month: 5, day: 13 }),
--
../calendars/portugal.mjs:117:      "key": "saintAnthonyOfLisbonPriestAndDoctorOfTheChurch",
../calendars/portugal.mjs-118-      "type": Types[4],
../calendars/portugal.mjs-119-      "moment": moment.utc({ year: year, month: 5, day: 13 }),
```

St Anthony of Padua and St Anthony of Lisbon is one and the same saint ([src](https://en.wikipedia.org/wiki/Anthony_of_Padua)).

**Comment by tukusejssirs** [comment]
$ grep -in matule enUs.mjs 
568:    "blessedJerzyMatulewiczBishop": "Blessed Jerzy Matulewicz, Bishop",
682:    "blessedGeorgeMatulewiczBishop": "Blessed George Matulewicz, Bishop",

**Comment by tukusejssirs** [comment]
$ grep -in "Motherofmercy\|dawn" enUs.mjs 
572:    "ourLadyMotherOfMercy": "Our Lady, Mother of Mercy",
692:    "ourLadyOfTheGateOfDawn": "Our Lady of The Gate of Dawn",

**Comment by tukusejssirs** [comment]
$ grep -in saintJohnOfTheCross enUs.mjs 
248:    "saintJohnOfTheCrossPriestAndDoctor": "Saint John of the Cross, Priest and Doctor",
723:    "saintJohnOfTheCrossDoctorOfTheChurch": "Saint John of The Cross, Doctor of The Church",

**Comment by tukusejssirs** [comment]
This issue is fixed except for the Sts Cyril and Methodius celebration, but that will be dealt in #117.

I’ll close this issue.
